Code split. Better testable structure. Unit tested. Coverage

hero101 opened the pull request:
The outcome is more structured, easy to maintain code which is also unit testable

<img width="755" height="48" alt="image" src="https://github.com/user-attachments/assets/136e9ee7-3523-4476-b031-a00ab3bbbea8" />

<img width="1889" height="920" alt="image" src="https://github.com/user-attachments/assets/ec7b27ab-b9ad-4096-8138-48dc3a75f745" />




<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added Alkemio authentication, authorization, and storage extensions; Sender service for integrations; Yjs v2 import/export utilities; connection-management with extension priority ordering.

- Refactor
  - Server wiring updated to use the new extensions and now exposes the underlying real-time instance.

- Tests
  - Extensive unit test suites added across extensions, server, integration, sender, transforms, and utilities.

- Documentation
  - Copilot usage guide and expanded architecture document added.

- Chores
  - Test/coverage/UI scripts and dev tooling updated; .coverage ignored; LF line-ending rules tightened.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hero101 committed:
wip

hero101 committed:
Normalize line endings to LF

hero101 committed:
read only reason

hero101 committed:
improved collaborator check

hero101 committed:
Merge branch 'memo-license' into improvements

hero101 committed:
improved git attributes for text

hero101 committed:
WIP - split auth

hero101 committed:
WIP - split auth; forgotten file

hero101 committed:
Merge branch 'develop' of https://github.com/alkem-io/collaborative-document-service into improvements

hero101 committed:
split authentication and authorization

hero101 committed:
split storage

hero101 committed:
commiting working tests

hero101 committed:
storage tests

hero101 committed:
authorizer tests

valentinyanakiev commented:
this is quite good but requires some work to be merged into dev / master - can you wrap it up?

hero101 committed:
wip

hero101 committed:
working tests

hero101 committed:
add copilot instructions for project guidelines and coding standards

hero101 committed:
added more test for integration.service.ts

hero101 committed:
added coverage settings & report

hero101 committed:
increased coverage on sender.service.ts

hero101 committed:
increased coverage on integration.service.ts

hero101 committed:
improved coverage config

hero101 committed:
fixed incorrect imports

hero101 committed:
fixed building errors

hero101 committed:
fixed nest dependency error

hero101 committed:
updated to latest vite with proper path resolver

hero101 committed:
Merge branch 'develop' of https://github.com/alkem-io/collaborative-document-service into improvements

# Conflicts:
#	package.json
#	pnpm-lock.yaml
#	src/hocuspocus/extensions/authentication/types/authentication.interfaces.ts
#	src/hocuspocus/hocuspocus.module.ts
#	src/hocuspocus/hocuspocus.server.ts
#	src/services/integration/integration.service.ts
#	test/utils/default.mocker.factory.ts

hero101 committed:
fixed broken and slow tests

hero101 committed:
authorization service tests

hero101 committed:
util.service.ts covered with tests

hero101 committed:
added tests for north.star.metric.extension.ts

hero101 committed:
added tests for hocuspocus.server.ts

hero101 committed:
fixed bug and increased coverage

hero101 committed:
increased coverage

hero101 commented:
@coderabbit pause

hero101 commented on src/hocuspocus/extensions/storage/alkemio-storage/alkemio.storage.ts:
```diff
@@ -0,0 +1,114 @@
+    super(AlkemioStorage.name);
+  }
+
+  /**
+   * Called once, when the first client connects to the server, during the creation of a new document.
+   * Called after onAuthenticate
+   */
+  public async onLoadDocument({
+    documentName: documentId,
+  }: onLoadDocumentPayload): Promise<Doc | null> {
+    try {
+      return await this.storageService.loadDocument(documentId);
+    } catch (error: any) {
+      this.logger.error(
+        {
+          message: 'Failed to load document.',
+          documentId,
+          error,
+        },
+        error?.stack,
+        LogContext.STORAGE
+      );
+      // Return null to let Hocuspocus create a new document
+      throw error;
+    }
+  }
```
I would like the server to throw and terminate the connection if a document that does not exist is fetched.

hero101 commented on test/config/vitest.config.ts:
```diff
@@ -1,16 +1,61 @@
+      '@test': path.resolve(__dirname, '../../test'),
+      '@common': path.resolve(__dirname, '../../src/common'),
+    },
+  },
```
suggest a diff update I should apply

hero101 replied:
@coderabbit generate me a diff to this comment to apply.

hero101 committed:
addressed automated comments

hero101 commented:
@coderabbitai resume

hero101 commented:
@coderabbit review

hero101 committed:
comments addressed

hero101 committed:
comments addressed

hero101 committed:
overall architectural overview documentation

hero101 committed:
Merge branch 'develop' of https://github.com/alkem-io/collaborative-document-service into improvements

hero101 committed:
refactor: replace email with user ID in authentication and authorization logs

hero101 committed:
fixed 6 failing tests

valentinyanakiev committed:
Logs errors during client proxy creation

Enhances robustness by capturing and reporting failures when initializing the client proxy. Previously, these errors were silently swallowed, making it difficult to diagnose setup issues.
